crates/apollo-compiler/tests/validation/field_merging.rs

```diff
@@ -0,0 +1,1394 @@
+                ────╯
+            "#]],
+        );
+    }
+
+    #[test]
+    fn dedupe_conflicts() {
+        expect_errors(
+            r#"
+      {
+        f1 {
+          ...A
+          ...B
+        }
+        f2 {
+          ...B
+          ...A
+        }
+        f3 {
+          ...A
+          ...B
+          x: c
+        }
+      }
+      fragment A on Type {
+        x: a
+      }
+      fragment B on Type {
+        x: b
+      }
+    "#,
+            expect![[r#"
+                Error: cannot select multiple fields into the same alias `x`
+                    ╭─[query.graphql:17:3]
+                    │
+                 17 │   x: a
+                    │   ──┬─  
+                    │     ╰─── `x` is selected from `Type.a` here
+                    │ 
+                 20 │   x: b
+                    │   ──┬─  
+                    │     ╰─── `x` is selected from `Type.b` here
+                    │ 
+                    │ Help: Both fields may be present on the schema type, so it's not clear which one should be used to fill the response
+                ────╯
+                Error: cannot select multiple fields into the same alias `x`
+                    ╭─[query.graphql:17:3]
+                    │
+                 13 │     x: c
+                    │     ──┬─  
+                    │       ╰─── `x` is selected from `Type.c` here
+                    │ 
+                 17 │   x: a
+                    │   ──┬─  
+                    │     ╰─── `x` is selected from `Type.a` here
+                    │ 
+                    │ Help: Both fields may be present on the schema type, so it's not clear which one should be used to fill the response
+                ────╯
+                Error: cannot select multiple fields into the same alias `x`
+                    ╭─[query.graphql:20:3]
+                    │
+                 17 │   x: a
+                    │   ──┬─  
+                    │     ╰─── `x` is selected from `Type.a` here
+                    │ 
+                 20 │   x: b
+                    │   ──┬─  
+                    │     ╰─── `x` is selected from `Type.b` here
+                    │ 
+                    │ Help: Both fields may be present on the schema type, so it's not clear which one should be used to fill the response
+                ────╯
+                Error: cannot select multiple fields into the same alias `x`
+                    ╭─[query.graphql:20:3]
+                    │
+                 13 │     x: c
+                    │     ──┬─  
+                    │       ╰─── `x` is selected from `Type.c` here
+                    │ 
+                 20 │   x: b
+                    │   ──┬─  
+                    │     ╰─── `x` is selected from `Type.b` here
+                    │ 
+                    │ Help: Both fields may be present on the schema type, so it's not clear which one should be used to fill the response
+                ────╯
```

Comment:
This test is specifically meant to check that we don't produce too many warnings, and I think this may be one too many (but it may not be). It would be more obvious to see if this is correct as is, if it included a `SelectionPath`